### DISCUSSION TITLE
Viewing results of pipeline steps

### DISCUSSION THREAD

**Discussion by cflist** [tektoncd/pipeline#3840] [Q&A]
Coming from a background of Jenkins and Gitlab-ci I want to be able to quickly see the build results such as the unit test and code quality reports when a tasks succeed or fails. I haven't been able to find a clear way as how one is supposed to do this. It could be this way of thinking doesn't fit in the tekton approach, but if so I would like to know the alternative. Two of the in my opinion not ideal solutions I found are:

- Using PipelineResources as StorageResource 
This does not seem ideal since the docs mention to only support GCS and PipelineResources will most likely be deprecated
- Create a task that uploads the results somewhere for you to view, e.g. s3 and returning the url as [result](https://tekton.dev/docs/pipelines/pipelines/#emitting-results-from-a-pipeline) output.

What is the intended way to deal with viewing a detailed result of why a pipeline task succeeded or failed?

**Comment by ghost** [comment]
The [Dashboard project](https://github.com/tektoncd/dashboard) provides a user interface which lets you see the step/task/pipeline success and error states, results, log output and so on.  We host a version of the dashboard in read-only mode for our own CICD flows. You can take a look around it here: https://dashboard.dogfooding.tekton.dev/#/pipelineruns
